application/features/bootstrap/MfaUnitTestsContext.php

```diff
@@ -49,19 +57,84 @@ protected function createNewUserInDatabase($username)
         return $user;
     }
 
+    protected function createMfa($type, $verified=1, $user=null)
+    {
+        if ($user ===null) {
+            $user = $this->tempUser;
+        }
+        $mfa = new Mfa();
+        $mfa->user_id = $user->id;
+        $mfa->type = $type;
+        $mfa->verified = $verified;
+
+        Assert::true($mfa->save(), "Could not create new mfa.");
+        $user->refresh();
+        $this->mfaId = $mfa['id'];
+        $this->mfa = Mfa::findOne(['id' => $mfa['id']]);
```

Comment:
It looks like this line...
```php
$this->mfa = Mfa::findOne(['id' => $mfa['id']]);
```
... is simply re-getting the Mfa that we just created. Would the following have worked?
```php
$this->mfa = $mfa;
```

Reply:
Done.